components/high-sale-restaurant.js

```diff
@@ -0,0 +1,57 @@
+import { ordersData } from '../services/zomato-services.js';
+
+
+
+/**
+ * Identify restaurant which did sell highest
+ * 
+ */
+export function getHighSaleRestaurant() {
+    const orders = ordersData;
+
+    orders.forEach(order => {
+    });
+    const restaurantSalesMap = new Map();
+
+    getRestaurantSalesMap(restaurantSalesMap, orders);
+
+    return getHighestSaleRestaurantDetails(restaurantSalesMap);
+
+}
+
+/**
+ * To get the restaurant and the total price
+ * @param {*} restaurantSalesMap 
+ * @param {*} orders 
+ */
+
+function getRestaurantSalesMap(restaurantSalesMap, orders) {
+    orders.forEach(order => {
+        const { restaurant_name, price } = order;
+
+        if (restaurantSalesMap.has(restaurant_name)) {
```

Comment:
could you let me know why map is used here, 
where as in other places reduce is used